app/main/views/questionnaire.py

```diff
@@ -195,8 +315,26 @@ def _render_template(location, context, rendered_schema_json=None, template=None
     metadata = get_metadata(current_user)
     answers = get_answers(current_user)
     metadata_context = build_metadata_context(metadata)
-    previous_location = get_navigator(g.schema_json).get_previous_location(answers, location)
+
+    navigator = Navigator(g.schema_json, get_answer_store(current_user))
+    previous_location = navigator.get_previous_location(current_block_id=location)
```

Comment:
It could, and I think previously would have without the group iteration behaviour before. Not sure why that method doesn't any longer. Will investigate.